python/mirascope/llm/clients/openai/client.py

```diff
@@ -107,22 +108,45 @@ def structured_call(
 
         format_info = _formatting_utils.ensure_formattable(format)
         mode = format_info.mode
+        model_lacks_structured_output_support = (
+            model in _utils.MODELS_WITHOUT_STRUCTURED_OUTPUT_SUPPORT
+        )
+        if model_lacks_structured_output_support and mode == "strict-or-tool":
+            logging.warning(
+                "Model %s does not support strict formatting; falling back to tool",
+                model,
+            )
+            mode = "tool"
+        elif model_lacks_structured_output_support and mode == "strict-or-json":
+            logging.warning(
+                "Model %s does not support strict formatting; falling back to json",
+                model,
+            )
+            mode = "json"
 
         instruction_parts: list[str] = []
         if mode == "tool":
             instruction_parts.append(
-                f"When you are ready to respond to the user, use the {FORMAT_TOOL_NAME} tool to construct a properly formatted response."
+                f"When you are ready to respond to the user, call the {FORMAT_TOOL_NAME} tool to output a structured response."
+            )
+            instruction_parts.append(
+                "Do NOT output any text in addition to the tool call."
             )
         elif mode == "json":
             instruction_parts.append(
                 _base_utils.create_json_mode_instructions(format_info)
             )
+            if model_lacks_structured_output_support:
+                instruction_parts.append(
+                    "Respond ONLY with valid JSON, and no other text."
+                )
         if format_info.formatting_instructions:
             instruction_parts.append(format_info.formatting_instructions)
         if instruction_parts:
             messages = _base_utils.add_system_instructions(
                 messages, "\n".join(instruction_parts)
             )
+
```

Comment:
Note: Newline structure here doesn't exactly match the pattern for the simpler calls, because with the added complexity here it became (imo) ugly and hard to read.

I think the clients are going to get refactored considerably once we finish implementing everything. I'm willing to do a newline consistency pass once the clients are feature complete, however until then I expect them to diverge somewhat.